src/android/ConnectPlugin.java

```diff
@@ -324,13 +322,30 @@ public void run() {
     }
 
     private void executeUserIsChild(JSONArray args, CallbackContext callbackContext) {
-        try {
-            isChild = args.getBoolean(0);
-        } catch (JSONException e) {
-            isChild = false;
-        }
+        cordova.getThreadPool().execute(new Runnable() {
+            public void run() {
+                Boolean value;
+                try {
+                    value = args.getBoolean(0);
+                } catch (JSONException e) {
+                    value = true;
+                }
+                setUserIsChild(value);
+            }
+        });
+    }
 
-        AdSettings.setIsChildDirected(isChild);
+    private void setUserIsChild(Boolean value) {
+        isChild = value;
+        AdSettings.setMixedAudience(isChild);
+        FacebookSdk.setAutoLogAppEventsEnabled(!isChild);
+        FacebookSdk.setAdvertiserIDCollectionEnabled(!isChild);
+        if (!isChild && !FacebookSdk.getAutoInitEnabled()) {
+            FacebookSdk.setAutoInitEnabled(true);
+            FacebookSdk.fullyInitialize();
```

Comment:
Will this get called twice if login as parent, then switch to child and back to Parent again?

Reply:
Might do, yes

Reply:
Doesn't seem to do anything untoward, though